File: src/main/java/com/ggardet/patient/spi/dto/MedicationDTO.java
package com.ggardet.patient.spi.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.util.UUID;

public record MedicationDTO(
        UUID id,
        @NotNull UUID patientId,
        @NotBlank String name,
        String dosage,
        String frequency,
        LocalDate startDate,
        LocalDate endDate,
        UUID prescribingDoctorId,
        String instructions
) {}

File: src/main/kotlin/com/ggardet/patient/spi/dto/MedicationDTO.kt
package com.ggardet.patient.spi.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDate
import java.util.*

data class MedicationDTO(
    val id: UUID? = null,
    @field:NotNull val patientId: UUID,
    @field:NotBlank val name: String,
    val dosage: String? = null,
    val frequency: String? = null,
    val startDate: LocalDate? = null,
    val endDate: LocalDate? = null,
    val prescribingDoctorId: UUID? = null,
    val instructions: String? = null
)